### PR TITLE
[FLINK-22198][connector/kafka] Redirect KafkaContainer output to log4j and print debug log if test hangs

### PR DESCRIPTION
**Pull request by PatrickRen**
## What is the purpose of the change

This pull request adds more logging utils for ```KafkaTableTestBase``` in order to reveal more information if test hangs.

## Brief change log
- Redirect KafkaContainer's STDOUT and STDERR to log4j
- Add debugging log if the test hangs for 5 minutes

## Verifying this change
This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java
Comment (xintongsong) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java:91]: Is it possible to achieve this by overriding `KafkaContainer#doStart()`?
What the current approach make me uncomfortable are:
- It makes `kafkaContainer` non-final, which lead to potential problems like NPEs or so.
- As an abstract class, `startTestEnv` and `tearDownTestEnv` could be overridden.

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java
Comment (xintongsong) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java:113]: What is the rationality behind the behavior that prints debug logs for once at the 5min timeout?
- If the test case is expected to finish within 5min, we should also fail is on timeout.
- If there's no assumption that the case should finish within 5min, we may print the logs before the test case stuck.

I'd suggest to simply probe the topics periodically, to decouple the logging logs from the testing logics.

**Review by xintongsong — CHANGES_REQUESTED** [review]

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java
Comment (PatrickRen) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java]: Fixed by d6bfdd9

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java
Comment (PatrickRen) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java]: Fixed by d6bfdd9

**Comment by PatrickRen** [comment]
Thanks for the review @xintongsong !  I made another push based on your comments. Also I leave these logging as INFO level for debugging FLINK-22198. Could you please have another review? Thanks!

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java
Comment (xintongsong) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java:148]: These methods can be `private`.
- scheduleTimeoutLogger
- cancelTimeoutLogger
- describeExternalTopics
- logTopicPartitionStatus

**Review by xintongsong — APPROVED** [review]
Thanks, @PatrickRen. LGTM.
I have only one minor comment, which I'll address myself.
I'm taking over from here, and will merge this once AZP gives green light.

**Pull request closed without merging**
